anitya/lib/backends/crates.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+from anitya.lib.backends import BaseBackend
+from anitya.lib.exceptions import AnityaPluginException
+
+"""
+ © 2017 Igor Gnatenko <ignatenkobrain@fedoraproject.org>
+
+"""
+
+
+class CratesBackend(BaseBackend):
+    ''' The crates class for projects hosted on crates.io. '''
+
+    name = 'crates.io'
+    examples = [
+        'https://crates.io/crates/clap',
+        'https://crates.io/crates/serde',
+    ]
+
+    @classmethod
+    def _get_versions(cls, project):
+        url = 'https://crates.io/api/v1/crates/%s/versions' % project.name
+        try:
+            req = cls.call_url(url)
+        except Exception:  # pragma: no cover
+            raise AnityaPluginException('Could not contact %r' % url)
+
+        try:
+            data = req.json()
+        except Exception:  # pragma: no cover
```

Comment:
Calling ``json()`` on a response that can't be decoded will raise a ``ValueError`` so we should explicitly catch that here rather than ``Exception``.